modules/frontend/combiner/search.go

```diff
@@ -0,0 +1,94 @@
+package combiner
+
+import (
+	"fmt"
+	"io"
+	"sort"
+
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/grafana/tempo/pkg/tempopb"
+)
+
+var _ Combiner = (*genericCombiner[*tempopb.SearchResponse])(nil)
+
+func NewSearch() Combiner {
```

Comment:
There is already a combiner in https://github.com/grafana/tempo/blob/main/pkg/traceql/combine.go, and it looks like the only reference is in the frontend also.  Let's compare and consolidate to this new one if possible.

Reply:
I look at it, and that combiner is slightly different in the way we merge spans. also that combiner doesn't lock, but this one does.

I will leave a TODO to consolidate them, but would prefer to keep them spereate in this PR :)